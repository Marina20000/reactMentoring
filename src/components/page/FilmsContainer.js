import React from 'react';
import Flexbox from 'flexbox-react';
import MovieItem from './../movieItems/MovieItem';

export default class FilmsContainer extends React.Component {
  constructor(props) {
    super(props);
  }


  render() {
    let images = [];
    let movies = !!this.props.movies?this.props.movies:[];
    movies.forEach(index => {
      images.push(<MovieItem params={index} key={index.id} />)
    })
    return (<Flexbox flexDirection="column" minHeight="100vh" width = "100%">
      <Flexbox element="header" height="60px">
      </Flexbox>
      <Flexbox flexGrow={1} flexWrap={'wrap'} justifyContent={'center'}>
        {images}
      </Flexbox>
      <Flexbox element="footer" height="60px">
      </Flexbox>
    </Flexbox>);
  };
}
