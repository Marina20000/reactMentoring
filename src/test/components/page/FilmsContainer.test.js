import React from "react";
import renderer from 'react-test-renderer';
import FilmsContainer from './../../../components/page/FilmsContainer';

describe('FilmsContainer ', () => {
    it("testing with snapshot", () => {
        const tree = renderer.create(<FilmsContainer />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})