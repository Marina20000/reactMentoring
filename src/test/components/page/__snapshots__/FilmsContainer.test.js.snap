// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsContainer  testing with snapshot 1`] = `
<div
  className="sc-bdVaJa ewctWx"
>
  <header
    className="sc-bdVaJa enORLM"
  />
  <div
    className="sc-bdVaJa dDiHWc"
  />
  <footer
    className="sc-bdVaJa enORLM"
  />
</div>
`;
